### PR TITLE
Fix rage fist counter triggering from substitute

### PR DESCRIPTION
**Pull request by aviettran**
Not super familiar with client code, but I think this should cover things like sub, belly drum, etc..

https://www.smogon.com/forums/threads/bug-reports-v4-read-original-post-before-posting.3663703/page-97#post-9431398

### CODE REVIEW

File: src/battle.ts [hunk 1]
```diff
@@ -1686,7 +1686,9 @@ export class Battle {
 					break;
 				}
 			} else {
-				poke.timesAttacked += 1;
+				if (Dex.moves.get(this.lastMove).category !== "Status") {
+					poke.timesAttacked += 1;
```
Comment (Karthik99999) [inline src/battle.ts:1690]:
```suggestion
					poke.timesAttacked++;
```

File: src/battle.ts [hunk 2]
```diff
@@ -1686,7 +1686,9 @@ export class Battle {
 					break;
 				}
 			} else {
-				poke.timesAttacked += 1;
+				if (Dex.moves.get(this.lastMove).category !== "Status") {
```
Comment (Karthik99999) [inline src/battle.ts:1689]:
```suggestion
				if (this.dex.moves.get(this.lastMove).category !== 'Status') {
```

**Comment by Karthik99999** [comment]
I think this is a pretty good solution

**Comment by aviettran** [comment]
thanks

**Pull request merged**